### DISCUSSION TITLE
ATtiny43U

### DISCUSSION THREAD

**Discussion by GoSamGo95** [SpenceKonde/ATTinyCore#699] [Support / Q & A]
Got the error below after I run bootloader. Can anyone assist me? 

avrdude: Expected signature for ATtiny43u is 1E 92 0C
         Double check chip, or use -F to override this check.
Error while burning bootloader.

**Comment by SpenceKonde** [comment]
Turn on verbose output while uploading, and it will tell you what the
invalid signature it saw was. Combinations with only 00 or ff bytes are
wiring problems or board set to use clock source that isn't connected.
0x000000 in particular is the expected result if you have it set to use a
nonexistent clock source.

If it's anything else, please share the error output with verbose errors
enabled

____________
Spence Konde
Azzy’S Electronics

New products! Check them out at tindie.com/stores/DrAzzy
GitHub: github.com/SpenceKonde
ATTinyCore: Arduino support for almost every ATTiny microcontroller
Contact: ***@***.***

**Reply by GoSamGo95** [reply]
Hello and thank you for your reponse. 

Below are the Verbose output error. As shown below are either 00 or FF. Does it meant my connections are wrong? 


avrdude: Version 6.3-20190619
         Copyright (c) 2000-2005 Brian Dean, http://www.bdmicro.com/
         Copyright (c) 2007-2014 Joerg Wunsch

         System wide configuration file is "C:\Users\samue\AppData\Local\Arduino15\packages\ATTinyCore\hardware\avr\1.3.3/avrdude.conf"

         Using Port                    : COM5
         Using Programmer              : stk500v1
         Overriding Baud Rate          : 19200
         Setting bit clk period        : 5.0
         AVR Part                      : ATtiny43u
         Chip Erase delay              : 1000 us
         PAGEL                         : P00
         BS2                           : P00
         RESET disposition             : possible i/o
         RETRY pulse                   : SCK
         serial program mode           : yes
         parallel program mode         : yes
         Timeout                       : 200
         StabDelay                     : 100
         CmdexeDelay                   : 25
         SyncLoops                     : 32
         ByteDelay                     : 0
         PollIndex                     : 3
         PollValue                     : 0x53
         Memory Detail                 :

                                  Block Poll               Page                       Polled
           Memory Type Mode Delay Size  Indx Paged  Size   Size #Pages MinW  MaxW   ReadBack
           ----------- ---- ----- ----- ---- ------ ------ ---- ------ ----- ----- ---------
           eeprom        65     5     4    0 yes        64    4     16  4000  4500 0xff 0xff
           flash         65    10    64    0 yes      4096   64     64  4500  4500 0xff 0xff
           signature      0     0     0    0 no          3    0      0     0     0 0x00 0x00
           lock           0     0     0    0 no          1    0      0  4500  4500 0x00 0x00
           lfuse          0     0     0    0 no          1    0      0  4500  4500 0x00 0x00
           hfuse          0     0     0    0 no          1    0      0  4500  4500 0x00 0x00
           efuse          0     0     0    0 no          1    0      0  4500  4500 0x00 0x00
           calibration    0     0     0    0 no          2    0      0     0     0 0x00 0x00

         Programmer Type : STK500
         Description     : Atmel STK500 Version 1.x firmware
         Hardware Version: 2
         Firmware Version: 1.18
         Topcard         : Unknown
         Vtarget         : 0.0 V
         Varef           : 0.0 V
         Oscillator      : Off
         SCK period      : 0.1 us

avrdude: AVR device initialized and ready to accept instructions

Reading | ################################################## | 100% 0.02s

avrdude: Device signature = 0x000000 (retrying)

Reading | ################################################## | 100% 0.02s

avrdude: Device signature = 0x000000 (retrying)

Error while burning bootloader.
Reading | ################################################## | 100% 0.03s

avrdude: Device signature = 0x000000
avrdude: Yikes!  Invalid device signature.
         Double check connections and try again, or use -F to override
         this check.


avrdude done.  Thank you.

**Reply by SpenceKonde** [reply]
Chip is connected incorrectly **or** the initial write attempt on the fuses (there's no bootloader, so you're just writing the fuses when you do burn bootloader, not writing any bootloader) succeeded at writing the fuses... but wrote the wrong value! If the clock selection fuse is set to use a clock source that isn't present, you read all 0's. 

If you have a fresh chip, and try to burn bootloader, you can burn bootloader once with verbose output. Copy the output (DO NOT USE THE COPY ERROR TEXT BUTTON, see below) to a text file, then attempt to burn bootloader once more, and again copy the output to a second text file, and compare the two logs either visually or with your preferred diff or merge tool. If you got no error from burn bootloader, but get an error uploading via ISP afterwards that points almost indisputably to a bug in the core. I don't know what version of the core you are using. 

The copy error text button on the ide has a bug that's showing itself off here: It omits the very first line of upload output, which is the text of the command that it is executing. The output from the verbose upload that is useful for debugging this issue is the 3 byte ID that it saw (0x000000 = problem with fuses, or maybe wiring, while 0xFFFFFF indicates almost certainly an error in wiring that involves Vdd and MISO being shorted, and a mix of F's and 0's is usually wiring, not fuses. Everything else in the verbose output is worthless (the numbers in that table of values are all bogus), except for the first line which tells me what settings were used for burn bootloader. 

But wait wait wait, is that v 1.3.3 of the core you are using? You should be using 1.5.2 (the newest in board manager) or 2.0.0-dev from github, but 1.3.3? Whatever bug you're tripping over is likely one that was fixed ages ago.
